### PR TITLE
Partially port FML to Windows.

### PR DESCRIPTION
**Pull request by goderbauer**
* Adds a message loop impl for Windows
* Ports `thread.cc` to Windows

All FML unittests are now passing on Windows.

FML as a whole does not compile on windows yet because `mapping.cc` imports `sys/mman.h`, which is not available on Windows and the replacement API for memory-mapped files is very different on Windows, see https://msdn.microsoft.com/en-us/library/windows/desktop/aa366556%28v=vs.85%29.aspx.

### CODE REVIEW

File: fml/platform/win/message_loop_win.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#include "flutter/fml/message_loop_impl.h"
+#include "lib/ftl/macros.h"
+
+namespace fml {
+
+class MessageLoopWin : public MessageLoopImpl {
+ private:
+  bool running_;
+  HANDLE timer_;
```
Comment (chinmaygarde) [inline fml/platform/win/message_loop_win.h]: Can we use `ftl/unique_object` for all raw handles please.

File: fml/thread.cc [hunk 2]
```diff
@@ -55,6 +72,17 @@ void Thread::SetCurrentThreadName(const std::string& name) {
   pthread_setname_np(name.c_str());
 #elif OS_LINUX || OS_ANDROID
   pthread_setname_np(pthread_self(), name.c_str());
+#elif OS_WIN
+  THREADNAME_INFO info;
+  info.dwType = 0x1000;
+  info.szName = name.c_str();
+  info.dwThreadID = GetCurrentThreadId();
+  info.dwFlags = 0;
+  __try {
```
Comment (chinmaygarde) [inline fml/thread.cc]: How does this work if we have disabled exceptions?

**Comment by chinmaygarde** [comment]
I realized we should also probably add a test that measures if there are any idle wakes in the run loop.

File: fml/thread.cc (hunk 2 above)
Comment (goderbauer) [inline fml/thread.cc]: It is my understanding that __try is a Microsoft extension to deal with SEH (https://msdn.microsoft.com/en-us/library/windows/desktop/ms680657(v=vs.85).aspx). It's separate from C++'s exception handling. SEH handling should not be affected by disabling C++ exceptions.

File: fml/platform/win/message_loop_win.h (hunk 1 above)
Comment (goderbauer) [inline fml/platform/win/message_loop_win.h]: Done.

**Pull request merged**